### PR TITLE
Show "Swipe action" hint only three times

### PR DESCRIPTION
**Pull request by wtimme**
With this branch, the hint is only shown once (#98). The show count is stored in `UserDefaults`.

The code contains a constant that controls the maximum number of times the hint should be shown. I build this with an `Integer` setting so that we can tweak it. If we decided to go with "Only once", we could also use a `Bool` instead, minimizing the writing operation to `UserDefaults`.

### CODE REVIEW

File: Cineaste/ViewController/SearchMovies/SearchMoviesViewController+SwipeHint.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        guard shouldAnimateShowSwipeActionHint else { return }
+
+        cell.animateSwipeHint()
+
+        rememberThatSwipeActionHintWasAnimated()
+    }
+
+    /// MARK: Persistency
+
+    private var swipeActionHintShowCountKey: String {
```
Comment (fbernutz) [inline Cineaste/ViewController/SearchMovies/SearchMoviesViewController+SwipeHint.swift]: We already write and read some values to/from UserDefaults. Could you have a look a the implementation of `PromtForReviewPersistence` or `UsernamePersistence` and add another one for something like `SwipeActionHintPersistence`? Maybe these enums are not a good idea anymore, as there are three already, but I really like to stay consistent for this.

**Comment by wtimme** [comment]
While doing a bit more of testing, I noticed an existing behaviour that seems like a bug:

1. Remove the app from the device (to reset `UserDefaults`)
2. Install and start the app
3. On the first screen, tap "Add movie"
4. Wait until the "trending" movies are displayed

**Expected behaviour:** The "slide action" hint is visible
**Observed behaviour:** The hint is not visible. Only after closing and opening the screen a second time shows the hint.

I had a look on `master` and the bug is present there as well.

File: Cineaste/ViewController/SearchMovies/SearchMoviesViewController+SwipeHint.swift (hunk 1 above)
Comment (wtimme) [reply Cineaste/ViewController/SearchMovies/SearchMoviesViewController+SwipeHint.swift]: Yes, consistency is definitely desirable. 👍🏼 I've added an `enum` accordingly.

**Comment by wtimme** [comment]
I agree with your conclusion. This is a good compromise. I've increased the "show count" to 3.

**Pull request merged**